pkg/controller/config/config_controller.go

```diff
@@ -264,6 +288,43 @@ func (r *ReconcileConfig) replayData(ctx context.Context, w *watch.Set) error {
 	return nil
 }
 
+// setDataExpectations sets expectations for the initial cached data set, based on the provided watch set.
+// Resources that cannot be listed will not be tracked.
+func (r *ReconcileConfig) setDataExpectations(ctx context.Context, instance *configv1alpha1.Config, w *watch.Set) {
+	for _, gvk := range w.Items() {
+		u := &unstructured.UnstructuredList{}
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   gvk.Group,
+			Version: gvk.Version,
+			Kind:    gvk.Kind + "List",
+		})
+		err := r.reader.List(ctx, u)
+		if err != nil {
```

Comment:
How do we avoid transient errors leading to skipping cache warming here?

Reply:
This is similar to your question above - how many errors is too many errors, how do we distinguish between a recoverable and an unrecoverable error, and how long do we want to hold up the controller in pursuit of the perfect.

Reply:
IMO the answer is more than one, as this is a tradeoff between availability and security and 1 is about as hard toward availability as we can go. Since we aim to improve security, a more conservative choice seems appropriate. The exact number is likely a judgement call.

The alternative is to discount error types we "know" are transient, but that might lead to the controller being perma-down if our assessment is inaccurate. However, one note in favor of the perma-down approach... if we assume that all non-recoverable errors are user-fixable (e.g. bad constraint template => delete that template), that may be a tradeoff some people are willing to take. I likely wouldn't choose it as a default, personally. 

The dynamics of how long it keeps the controller down... what are the levers for controlling this? Do we have control over the retry wait periods? Assuming we use retries, can we make the tolerances user tune-able?